R/learner_BART_surv_bart.R

```diff
@@ -0,0 +1,293 @@
+#' @title Survival Bayesian Additive Regression Trees Learner
+#' @author bblodfon
+#' @name mlr_learners_surv.bart
+#'
+#' @description
+#' Fits a Bayesian Additive Regression Trees (BART) learner to right-censored
+#' survival data.
+#' For prediction, we return the mean posterior estimates of the survival
+#' function and the corresponding `crank` (expected mortality) using
+#' [mlr3proba::.surv_return].
+#' The full posterior estimates are currently stored in the
+#' `learner$state$surv_test` slot, along with the number of test observations
+#' `N`, number of unique times in the train set `K` and number of posterior
+#' draws `M`.
+#' See example for more details.
+#'
+#' Calls [BART::mc.surv.bart()] from \CRANpkg{BART}.
+#' Based on BART version 2.9.4 (2023-03-25).
+#'
+#' @section Custom mlr3 defaults:
+#' - `mc.cores` (in general use as many as possible if no issues arise):
+#'   - Actual default: 2
+#'   - Adjusted value: 1
+#'   - Reason for change: May conflict with parallelization via \CRANpkg{future}.
+#'
+#' @section Initial parameter values:
```

Comment:
```suggestion
#' @section Custom mlr3 parameter:
```

Reply:
I added after this commit so it's okay now.